src/In.ProjectEKA.HipService/OpenMrs/OpenMrsClient.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using In.ProjectEKA.HipService.Logger;
+
+namespace In.ProjectEKA.HipService.OpenMrs
+{
+    public class OpenMrsClient: IOpenMrsClient
+    {
+        private readonly HttpClient httpClient;
+        private readonly OpenMrsConfiguration configuration;
+
+        public OpenMrsClient(HttpClient httpClient, OpenMrsConfiguration openMrsConfiguration)
+        {
+            this.httpClient = httpClient;
+            configuration = openMrsConfiguration;
+
+            SettingUpHeaderAuthorization();
+        }
+
+        public async Task<HttpResponseMessage> GetAsync(string openMrsUrl)
+        {
+            try {
+                return await httpClient.GetAsync(Path.Join(configuration.Url, openMrsUrl));
+            }
+            catch (Exception exception)
+            {
+                Log.Error(exception, exception.StackTrace);
+                throw exception;
```

Comment:
Codacy found an issue: [Consider using 'throw;' to preserve the stack trace.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5933330)